pkg/integration/components/viewDriver.go

```diff
@@ -98,13 +102,25 @@ func (self *ViewDriver) SelectedLine(matcher *matcher) *ViewDriver {
 // asserts on the index of the selected line. 0 is the first index, representing the line at the top of the view.
 func (self *ViewDriver) SelectedLineIdx(expected int) *ViewDriver {
 	self.t.assertWithRetries(func() (bool, string) {
-		actual := self.getView().SelectedLineIdx()
+		actual, ok := self.selectedLineIdxInPatchExplorer()
+		if !ok {
+			actual = self.getView().SelectedLineIdx()
+		}
 		return expected == actual, fmt.Sprintf("%s: Expected selected line index to be %d, got %d", self.context, expected, actual)
 	})
 
 	return self
 }
 
+func (self *ViewDriver) selectedLineIdxInPatchExplorer() (int, bool) {
+	context := self.t.gui.ContextForView(self.getView().Name())
+	patchExplorerContext, ok := context.(types.IPatchExplorerContext)
+	if ok && patchExplorerContext.GetState() != nil {
+		return patchExplorerContext.GetState().GetSelectedLineIdx(), true
+	}
+	return 0, false
+}
+
```

Comment:
As I'm still struggling with getting a handle on lazygit's architecture, it's entire possible that there's a nicer/cleaner way of doing this. Suggestions welcome.

I did consider the alternative of keeping the view's cursor position in sync with the state's selectedLineIdx, but I abandoned the idea because it seems to be a lot of work, it's hard to verify that it's correct at all times, and it just doesn't seem worth it.